pkg/apis/extensions/v1beta1/conversion.go

```diff
@@ -93,6 +96,70 @@ func Convert_v1_PodSpec_To_api_PodSpec(in *v1.PodSpec, out *api.PodSpec, s conve
 	return v1.Convert_v1_PodSpec_To_api_PodSpec(in, out, s)
 }
 
+func convertToMapSelector(selector *string) (map[string]string, error) {
+	p := labels.NewParser(labels.NewLexer(*selector, 0))
+	reqs, err := p.Parse()
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse the selector: %v", err)
+	}
+	mapSelector := make(map[string]string)
+	for _, req := range reqs {
+		if req.Operator() != labels.EqualsOperator {
+			return nil, fmt.Errorf("only map-based requirements can be converted")
+		}
+		vals := req.Values()
+		if vals.Len() != 1 {
+			return nil, fmt.Errorf("equals operator must have exactly one value")
+		}
+		val, ok := vals.PopAny()
+		if !ok {
+			return nil, fmt.Errorf("equals operator must have exactly one value")
+		}
+		mapSelector[req.Key()] = val
+	}
+	return mapSelector, nil
+}
+
+func Convert_extensions_ScaleStatus_To_v1beta1_ScaleStatus(in *extensions.ScaleStatus, out *ScaleStatus, s conversion.Scope) error {
+	if defaulting, found := s.DefaultingInterface(reflect.TypeOf(*in)); found {
+		defaulting.(func(*extensions.ScaleStatus))(in)
+	}
+	out.Replicas = int32(in.Replicas)
+	if in.Selector != nil {
+		out.TargetSelector = new(string)
+		*out.TargetSelector = *in.Selector
+		mapSelector, err := convertToMapSelector(out.TargetSelector)
+		if err == nil {
+			out.Selector = mapSelector
+		}
+	} else {
+		out.Selector = nil
+		out.TargetSelector = nil
+	}
+	return nil
+}
+
+func Convert_v1beta1_ScaleStatus_To_extensions_ScaleStatus(in *ScaleStatus, out *extensions.ScaleStatus, s conversion.Scope) error {
+	if defaulting, found := s.DefaultingInterface(reflect.TypeOf(*in)); found {
+		defaulting.(func(*ScaleStatus))(in)
+	}
+	out.Replicas = int(in.Replicas)
+	if in.TargetSelector != nil {
```

Comment:
This needs a comment.

```
// Normally when 2 fields map to the same internal value we favor the old field, since
// old clients can't be expected to know about new fields but clients that know about the
// new field can be expected to know about the old field (though that's not quite true, due
// to kubectl apply). However, these fields are readonly, so any non-nil value should work.
```